src/runtime/PromiseObject.cpp

```diff
@@ -325,4 +326,69 @@ Value promiseAllResolveElementFunction(ExecutionState& state, Value thisValue, s
     }
     return Value();
 }
+
+Value promiseThenFinally(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
+{
+    // Let onFinally be F.[[OnFinally]].
+    Value onFinally = argv[1];
+    Value result;
+
+    // Assert: IsCallable(onFinally) is true.
+    // Let result be ? Call(onFinally, undefined).
+    ASSERT(onFinally.asPointerValue()->isCallable());
+    result = Object::call(state, onFinally, Value(), 0, &thisValue);
+
+    // Let C be F.[[Constructor]].
+    // Assert: IsConstructor(C) is true.
+    Value C = argv[0];
+    ASSERT(C.asPointerValue()->isCallable());
+
+    Value promise = promiseResolve(state, C.asObject(), result);
+
+    // Let valueThunk be equivalent to a function that returns value.
+    // Return ? Invoke(promise, "then", « valueThunk »).
+    Object* thisObject = thisValue.toObject(state);
+    auto strings = &state.context()->staticStrings();
+    Value then = thisObject->get(state, strings->then).value(state, thisObject);
+    Value argument[1] = { Value(valueThunkHelper(argv[3])) };
+
+    // Let resolveElement be a new built-in function object as defined in Promise.all Resolve Element Functions.
+    FunctionObject* resolveElement = new NativeFunctionObject(state, NativeFunctionInfo(strings->Empty, promiseAllResolveElementFunction, 1, NativeFunctionInfo::Strict));
```

Comment:
I think this not part of the specification, also the the variable is not used at all.